5/3 (#398)
helium/HIP: 0052-lorawan-dao.md

# HIP 52: LoRaWAN subDAO

- Authors: [@tjain-mcc](https://github.com/tjain-mcc), [@shayons297](htts://github.com/shayons297), [jamiew](https://github.com/jamiew), [abhay](https://github.com/abhay)
- Start Date: 2022-01-04
- Category: Economic, Technical
- Original PR: https://github.com/helium/HIP/pull/335
- Tracking Issue: https://github.com/helium/HIP/issues/338
- Status: Draft

**Summary**

> In [HIP 51: Helium DAOs](https://docs.google.com/document/d/1ibFE2DI8fkd4uOSnIT00C7JiRYsZExg164_KtDzz-YI/edit#), we provide a general structure for onboarding new Decentralized Network Protocols (DNPs) to the broader Helium Network, with mechanisms in place to ensure that protocol-specific attributes such as proof-of-coverage rules and data transfer pricing are within control of the DNT DAOs.

Reviewer: ```suggestion
In [HIP 51: Helium DAOs](https://github.com/helium/HIP/blob/main/0051-helium-dao.md), we provide a general structure for onboarding new Decentralized Network Protocols (DNPs) to the broader Helium Network, with mechanisms in place to ensure that protocol-specific attributes such as proof-of-coverage rules and data transfer pricing are within control of the DNT DAOs.
```

5. **Proof-of-Coverage Mechanism** Most subnetworks will utilize a Proof of Coverage algorithm to verify on an ongoing basis that hotspots are accurately representing their location and the wireless network coverage they are creating from that location. Responsibilities here include Proof-of-Coverage challenge construction, target selection, reward scaling, and verification. Note that subnetworks can choose to skip this mechanism, but must provide reasoning in their subDAO proposal as to why it is not necessary for proper functioning of the subnetwork.
6. **Governance Structure** subDAOs retain control over critical components of the network, and subDAO members can propose and vote for changes to core parameters and mechanisms. Responsibilities here include specification of a formal on-chain voting process that is resistant to attacks.

The remainder of this proposal defines initial values for the subDAO given the aforementioned set of responsibilities.

### *Emissions Curve*

> There will be a max supply of 200B HLT.

Reviewer: ```suggestion
There will be a max supply of 200,000,000,000 HLT.
```

Users can choose to delegate their veHLT for three core purposes:

1. *Hotspot Self-Staking*: Hotspots are able to delegate veHLT in their own name in order to increase the frequency of challenges to that hotspot.

Users are able to stake either “for” or “against” hotspots. At the end of each epoch, hotspots have a net veHLT score in which “for” staked amounts are added, and “against” staked amounts are subtracted from the running total.

Hotspot veHLT scores are considered within a range of -10M and 10M. For the maximum amount of 10M veHLT staked, hotspots receive 10x the probability of being selected for a challenge within an epoch. For the default amount of zero veHLT staked, hotspots receive 1x the probability of being selected for a challenge within an epoch. For the minimum amount of -10M, hotspots receive zero challenges. In either direction across zero from the y-axis, the challenge probability multiplier varies linearly with the veHLT staked.

Users that delegate veHLT to hotspots receive 50% of any incremental earnings through increased challenge completion.

> ![https://s3.us-west-2.amazonaws.com/secure.notion-static.com/d60c7241-a481-407a-b58e-fceca4692de2/Screen_Shot_2022-05-04_at_10.35.07_AM.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Content-Sha256=UNSIGNED-PAYLOAD&X-Amz-Credential=AKIAT73L2G45EIPT3X45%2F20220504%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Date=20220504T143702Z&X-Amz-Expires=86400&X-Amz-Signature=503709360231f900a0de4d000135a39ff4a84bef9d1e9791882244cf4d8fd5eb&X-Amz-SignedHeaders=host&response-content-disposition=filename%20%3D%22Screen%2520Shot%25202022-05-04%2520at%252010.35.07%2520AM.png%22&x-id=GetObject](https://s3.us-west-2.amazonaws.com/secure.notion-static.com/d60c7241-a481-407a-b58e-fceca4692de2/Screen_Shot_2022-05-04_at_10.35.07_AM.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Content-Sha256=UNSIGNED-PAYLOAD&X-Amz-Credential=AKIAT73L2G45EIPT3X45%2F20220504%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Date=20220504T143702Z&X-Amz-Expires=86400&X-Amz-Signature=503709360231f900a0de4d000135a39ff4a84bef9d1e9791882244cf4d8fd5eb&X-Amz-SignedHeaders=host&response-content-disposition=filename%20%3D%22Screen%2520Shot%25202022-05-04%2520at%252010.35.07%2520AM.png%22&x-id=GetObject)

Reviewer: ```suggestion
![challenge-multiplier-vs-vehnt-delegated](https://s3.us-west-2.amazonaws.com/secure.notion-static.com/d60c7241-a481-407a-b58e-fceca4692de2/Screen_Shot_2022-05-04_at_10.35.07_AM.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Content-Sha256=UNSIGNED-PAYLOAD&X-Amz-Credential=AKIAT73L2G45EIPT3X45%2F20220504%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Date=20220504T143702Z&X-Amz-Expires=86400&X-Amz-Signature=503709360231f900a0de4d000135a39ff4a84bef9d1e9791882244cf4d8fd5eb&X-Amz-SignedHeaders=host&response-content-disposition=filename%20%3D%22Screen%2520Shot%25202022-05-04%2520at%252010.35.07%2520AM.png%22&x-id=GetObject)
```

![https://lh6.googleusercontent.com/nzEaWc5B055nckKgEIv-kFyNwP9dG3O2pZIfA_yJlwM6Q4XpqP-RTWA9lehb1yXRV7aRD-U0Isq_0p8hYHEyzDU_J6UHOB_sk5skER-k04VGsQ3DrDeKGnJdiWw86zWvHNX8qWCA](https://lh6.googleusercontent.com/nzEaWc5B055nckKgEIv-kFyNwP9dG3O2pZIfA_yJlwM6Q4XpqP-RTWA9lehb1yXRV7aRD-U0Isq_0p8hYHEyzDU_J6UHOB_sk5skER-k04VGsQ3DrDeKGnJdiWw86zWvHNX8qWCA)

At epoch T, we denote the value of H, S, and P as HT, ST,  and PT. The programmatic treasury formula at epoch T is a function in two variables, HT, ST defined as follows:

![https://lh3.googleusercontent.com/7P_5Bha1KNPmQl2wWBHPoB6ojHfFKdnSaD208DR1vFUE2ybnAopqA6X-sVMzhHqiH5DeeDDKMY5hxKOe74j_PVcEBCoXjwetlytpeS241lgrEy_TmapfPiWUjEzyhp72kyvLiEJA](https://lh3.googleusercontent.com/7P_5Bha1KNPmQl2wWBHPoB6ojHfFKdnSaD208DR1vFUE2ybnAopqA6X-sVMzhHqiH5DeeDDKMY5hxKOe74j_PVcEBCoXjwetlytpeS241lgrEy_TmapfPiWUjEzyhp72kyvLiEJA)

For example, consider at epoch #125, the total outstanding supply of tokens for the LoRaWAN subDAO is 3.275B HLT and the total amount in the treasury reserve is 2M HNT. The programmatic treasury formula is given as follows

> ![https://lh6.googleusercontent.com/J_34HUWBuZl_YxNOnMv_fVkgvJKVeWtNLdZ9eZfKjYOFVjvv9LqjyB-pMmBaqpgFykNb4YcKfE7CuHDC6aX1WQORR2vlyyg4hyb_38ACZqDMjUrwVKHUhYeZYVj1QYceVwL-z3gR](https://lh6.googleusercontent.com/J_34HUWBuZl_YxNOnMv_fVkgvJKVeWtNLdZ9eZfKjYOFVjvv9LqjyB-pMmBaqpgFykNb4YcKfE7CuHDC6aX1WQORR2vlyyg4hyb_38ACZqDMjUrwVKHUhYeZYVj1QYceVwL-z3gR)

Reviewer: ```suggestion
![example-redemption](https://lh6.googleusercontent.com/J_34HUWBuZl_YxNOnMv_fVkgvJKVeWtNLdZ9eZfKjYOFVjvv9LqjyB-pMmBaqpgFykNb4YcKfE7CuHDC6aX1WQORR2vlyyg4hyb_38ACZqDMjUrwVKHUhYeZYVj1QYceVwL-z3gR)
```

> This implies that at epoch #125, any number of HLT tokens can be redeemed at a unit price of 0.00061068702 HNT.

Reviewer: ```suggestion
This implies that at epoch #125, 1 HLT can be redeemed for 0.00061068702 HNT.
```

Now consider at epoch #225, the total outstanding supply of tokens for the LoRaWAN subDAO is 3.5B HLT and the total amount in the treasury reserve is 2.025M HNT. The programmatic treasury formula is given as follows

> ![https://lh5.googleusercontent.com/JzoHFD4OcK-YbK8aqiR2VjsBtN8mxRK6oEae_57KhoeoXp1TGKbRWGF7NkfRByITdTcfEAAkPSkKPMfz604aTcqhNlBJ2o7YJ6TjpAUxvGJSBcUgyH4o8iDmJRQ8ejcqaa8kvK9e](https://lh5.googleusercontent.com/JzoHFD4OcK-YbK8aqiR2VjsBtN8mxRK6oEae_57KhoeoXp1TGKbRWGF7NkfRByITdTcfEAAkPSkKPMfz604aTcqhNlBJ2o7YJ6TjpAUxvGJSBcUgyH4o8iDmJRQ8ejcqaa8kvK9e)

Reviewer: ```suggestion
![example-redemption-2](https://lh5.googleusercontent.com/JzoHFD4OcK-YbK8aqiR2VjsBtN8mxRK6oEae_57KhoeoXp1TGKbRWGF7NkfRByITdTcfEAAkPSkKPMfz604aTcqhNlBJ2o7YJ6TjpAUxvGJSBcUgyH4o8iDmJRQ8ejcqaa8kvK9e)
```

> This implies that at 100 epochs after #125 at #225, any number of HLT tokens can be redeemed at a unit price of 0.00057857142 HNT.

Reviewer: ```suggestion
This implies that at 100 epochs after #125 at #225, 1 HLT tokens can be redeemed for 0.00057857142 HNT.
```

In epoch #225, the total outstanding supply of HLT has increased, and given inflows to the treasury reserve in HNT the resulting unit price has decreased. If the rate at which HNT flows into the treasury reserve outpaces the emission of HLT tokens, the unit price of HLT increases.

Between epoch #125 and #225, a similar repricing takes place in each individual epoch given the amount of HNT in the treasury reserve and the outstanding supply of HLT tokens.

It is possible to represent the programmatic treasury formula as a surface in euclidean space using the following vector notation in two variables:

> ![https://lh3.googleusercontent.com/PNzok9G-6EnVEVVWri06NcbbRffoF_f9M5eHo9ain25TT4OoG9zAFnIAowTtWqTNyI9RFC98lICBCONmQqS6uqlQXpDcXXzieywg1IXizx9YgLpzAtOUGTvDXde5Qb0U6ds-7smt](https://lh3.googleusercontent.com/PNzok9G-6EnVEVVWri06NcbbRffoF_f9M5eHo9ain25TT4OoG9zAFnIAowTtWqTNyI9RFC98lICBCONmQqS6uqlQXpDcXXzieywg1IXizx9YgLpzAtOUGTvDXde5Qb0U6ds-7smt)

Reviewer: ```suggestion
![programmatic-treasury-formula](https://lh3.googleusercontent.com/PNzok9G-6EnVEVVWri06NcbbRffoF_f9M5eHo9ain25TT4OoG9zAFnIAowTtWqTNyI9RFC98lICBCONmQqS6uqlQXpDcXXzieywg1IXizx9YgLpzAtOUGTvDXde5Qb0U6ds-7smt)
```

Where

> ![https://lh6.googleusercontent.com/UqiNgjY8LVN6lzT1zgM4D-J3GJfxAeYomOjFirG4YION8tF0ZvrJmWQWKl2kL6XTAPWUhGMJnv_SZHIhXyvST7TJy4vm9AuRUy_a686QbT1gLoHpwhTZauml3G20FBw_qM1XCeX6](https://lh6.googleusercontent.com/UqiNgjY8LVN6lzT1zgM4D-J3GJfxAeYomOjFirG4YION8tF0ZvrJmWQWKl2kL6XTAPWUhGMJnv_SZHIhXyvST7TJy4vm9AuRUy_a686QbT1gLoHpwhTZauml3G20FBw_qM1XCeX6)

Reviewer: ```suggestion
![programmatic-treasury-formula-components](https://lh6.googleusercontent.com/UqiNgjY8LVN6lzT1zgM4D-J3GJfxAeYomOjFirG4YION8tF0ZvrJmWQWKl2kL6XTAPWUhGMJnv_SZHIhXyvST7TJy4vm9AuRUy_a686QbT1gLoHpwhTZauml3G20FBw_qM1XCeX6)
```

Clarifications:

- The supply of HLT tokens for the purposes of calculating the HLT/HNT price includes all locked veHLT tokens.
- The subDAO updates the limit order price every epoch **before** any new HLT are distributed out to stakeholders for their activity during that epoch.

### *Oracle Operations*

LoRaWAN oracles confirm proof of coverage, data transfer, and add blocks to the LoRaWAN subnetwork. They serve state data around Proof-of-Coverage challenges and data transfer events to light and data only hotspots.

Validation is performed by a set of rotating nodes known as the consensus group, which verifies transactions and ordering prior to forming a block and proposing it to the subnetwork chain. Consensus groups are elected once per epoch, and the number of members is given by the num_consensus_members chain variable (currently set at 43).

> *The Helium Consensus Protocol is based on a variant of the HoneyBadgerBFT (HBBFT) protocol. HBBFT is based on a body of research originally kicked off by Andrew Miller and the team at the University of Illinois, Urbana-Champaign.*

Reviewer: ```suggestion
The Helium Consensus Protocol is based on a variant of the HoneyBadgerBFT (HBBFT) protocol. HBBFT is based on a body of research originally kicked off by Andrew Miller and the team at the University of Illinois, Urbana-Champaign.
```

> *HBBFT is an asynchronous atomic broadcast protocol designed to enable a group of known nodes to achieve consensus over unreliable links. In Helium’s implementation, a consensus group of [elected Validators](https://explorer-beta.helium.com/validators) receives encrypted transactions as inputs and proceeds to reach common agreement on the ordering of these transactions before forming a block and adding it to the blockchain.*

Reviewer: ```suggestion
*HBBFT is an asynchronous atomic broadcast protocol designed to enable a group of known nodes to achieve consensus over unreliable links. In Helium’s implementation, a consensus group of [elected Validators](https://explorer.helium.com/validators) receives encrypted transactions as inputs and proceeds to reach common agreement on the ordering of these transactions before forming a block and adding it to the blockchain.*
```

> *HBBFT relies on a scheme known as threshold encryption. Using this scheme, transactions are encrypted using a shared public key, and are only decryptable when the elected consensus group works together to decrypt them. The usage of threshold encryption enables the Helium Consensus Protocol to achieve censorship-resistant transactions.*

Reviewer: ```suggestion
HBBFT relies on a scheme known as threshold encryption. Using this scheme, transactions are encrypted using a shared public key, and are only decryptable when the elected consensus group works together to decrypt them. The usage of threshold encryption enables the Helium Consensus Protocol to achieve censorship-resistant transactions.
```

> *At the end of each epoch, mining rewards are distributed by the consensus group to the wallet addresses that have earned them.*

Reviewer: ```suggestion
* At the end of each epoch, mining rewards are distributed by the consensus group to the wallet addresses that have earned them.
```

> *Each one of the above activities is recorded in a block using the reward transaction. At the completion of each epoch, all the individual reward transactions are grouped in a rewards transaction at which point all HNT mined in that epoch are distributed.*

Reviewer: ```suggestion
*Each one of the above activities is recorded in a block using the reward transaction. At the completion of each epoch, all the individual reward transactions are grouped in a rewards transaction at which point all HNT mined in that epoch are distributed.
```

2. Transmitter - Sometimes called "Challengee". This Hotspot is the target of the POC challenge and is responsible for transmitting (or "beaconing") challenge packets to potentially be witnessed by geographically proximate Hotspots.
3. Witness - Hotspots that are geographically proximate to the Transmitter and report the existence of the challenge packet after it has been transmitted.

### *Data Transfer and Pricing Specification*

Data Credits are utilized in asserting new hotspots and their location on the chain, registering OUIs and associated devices, and as payment for packet transfers.

With the activation of [HIP 10](https://github.com/helium/HIP/blob/master/0010-usage-based-data-transfer-rewards.md), hotspot operators receive HNT emissions up to 32.5% per epoch and are rewarded at 1:1 rate based on dollar value of Data Credits transfers as per the [HNT Price Oracle](https://docs.helium.com/blockchain/oracles). This proposal scales Data Credits rewards based on actual activity on the network, and disincentivizes arbitrageurs from taking advantage of more arbitrary distribution mechanisms for rewards.

> # [HIP 37](https://github.com/helium/HIP/blob/master/0037-omni-protocol-poc.md) proposed the removal of a division between proof-of-coverage and data credits rewards entirely at the date of the second HNT halving (8/1/2023). We propose this removal happen from the launch of the Lorawan subDAO. As shown in the emissions schedule, this is reflected in the HLT subDAO.

Reviewer: ```suggestion
[HIP 37](https://github.com/helium/HIP/blob/master/0037-omni-protocol-poc.md) proposed the removal of a division between proof-of-coverage and data credits rewards entirely at the date of the second HNT halving (8/1/2023). We propose this removal happen from the launch of the Lorawan subDAO. As shown in the emissions schedule, this is reflected in the HLT subDAO.
```

At present, every 24 bytes sent in a packet cost 1 DC, priced at $0.00001. Data Credits are generated by burning the requisite amount of HNT as per the Helium price oracle.

The transaction is as follows:

1. A Miner receives a packet and determines who it belongs to
2. The Miner reaches the appropriate Helium LoRaWAN Network Server and offers the packet

> 3. The Helium LoRaWAN Network Server expresses interest in the packet and accepts delivery

Reviewer: ```suggestion
3. The Helium Router expresses interest in the packet and accepts delivery
```
